genesyscloud/outbound_ruleset/resource_genesyscloud_outbound_ruleset.go

```diff
@@ -131,3 +142,38 @@ func deleteOutboundRuleset(ctx context.Context, d *schema.ResourceData, meta int
 		return retry.RetryableError(fmt.Errorf("Outbound Ruleset %s still exists", d.Id()))
 	})
 }
+
+// filterOutboundRulesets filters rule sets by removing the ones that reference skills that no longer exist in GC
+func filterOutboundRulesets(ruleSets []platformclientv2.Ruleset, skillMap resourceExporter.ResourceIDMetaMap) ([]platformclientv2.Ruleset, diag.Diagnostics) {
+	var filteredRuleSets []platformclientv2.Ruleset
+	log.Printf("Filtering outbound rule sets")
+
+	for _, ruleSet := range ruleSets {
+		var foundDeleted bool
+
+		for _, rule := range *ruleSet.Rules {
+			var err error
+
+			if foundDeleted, err = doesRuleActionsRefDeletedSkill(rule, skillMap); err != nil {
```

Comment:
@williamjeuniaux Looks better.  I do have a question though could the two conditions be anded together.  If I am reading this correctly you are breaking, the doesRuleActionsRef and doesRuleConditionsRef both cause a break in the loop.  Why not do an and in a single if and break if either is found true.

It just seems like we could simplify this code a little bit.  Sorry not trying to bit nitpicky

Reply:
That's ok! The idea is I didn't want to evaluate the conditions at all if there was a deleted skill in the actions. If i add an AND, i believe it will evaluate both.

I could add an OR, so it only evaluates conditions if actions returns false, but then I am getting another problem; it considers the expression as a single value context. Since i am returning two values (bool + error)  the compiler complains.

I could get rid of the error handling and just return a bool, but i am not a huge fan of this idea. Thoughts?